server/weaverbird/backends/sql_translator/steps/utils/query_transformation.py

```diff
@@ -43,10 +43,12 @@
 def apply_condition(condition: Condition, query: str) -> str:
     if isinstance(condition, ComparisonCondition):
         try:
-            if isinstance(condition.value, datetime.datetime):
+            if isinstance(condition.value, datetime.datetime) or isinstance(
+                condition.value, datetime.date
+            ):
                 query += (
-                    f'{condition.column} {SQL_COMPARISON_OPERATORS[condition.operator]} '
-                    f'to_timestamp(\'{condition.value}\')'
+                    f'to_timestamp({condition.column}) {SQL_COMPARISON_OPERATORS[condition.operator]} '
+                    f'to_timestamp(\'{condition.value.isoformat()}\')'
```

Comment:
Good idea to turn both to timestamps! I guess it could also work without but that should definitely work :+1: